lib/cgi/cookie.rb

```diff
@@ -123,6 +132,14 @@ def secure=(val)
       @secure
     end
 
+    # Set whether the Cookie is a httponly cookie or not.
+    #
+    # +val+ must be a boolean.
+    def httponly=(val)
+      @httponly = val if val == true or val == false
```

Comment:
I'm not sure if this check (along with #secure=) makes any sense.  I suppose it should raise a TypeError or ArgumentError if it doesn't accept a non-boolean value, instead of silently ignoring a value given.